modules/platforms/cpp/core/include/ignite/impl/cluster/cluster_node_impl.h

```diff
@@ -44,27 +45,189 @@ namespace ignite
                 /**
                  * Constructor used to create new instance.
                  *
-                 * @param reader Binary reader.
+                 * @param mem Memory to read Cluster Node.
                  */
-                ClusterNodeImpl(binary::BinaryReaderImpl& reader);
+                ClusterNodeImpl(common::concurrent::SharedPointer<interop::InteropMemory> mem);
 
                 /**
                  * Destructor.
                  */
                 ~ClusterNodeImpl();
 
+                /**
+                 * Get collection of addresses this node is known by.
+                 *
+                 * @return Collection of addresses this node is known by.
+                 */
+                std::vector<std::string> GetAddresses();
+
+                /**
+                 * Get a node attribute.
+                 *
+                 * @param name Node attribute name.
+                 * @return Node attribute.
+                 *
+                 * @throw IgniteError in case of attribute name does not exist
+                 * or if template type is not compatible with attribute.
+                 */
+                template<typename T>
+                T GetAttribute(std::string name)
```

Comment:
Let's add `bool IsAttributeSet() const` method for user so they will be able check if the attribute present in a simple way.